src/main/java/com/driving/school/api/StudentApi.java

```diff
@@ -0,0 +1,36 @@
+package com.driving.school.api;
+
+import com.driving.school.model.StudentModel;
+import com.driving.school.service.StudentService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@CrossOrigin(origins = "http://localhost:4200")
+@RequestMapping(value = "/students")
+public class StudentApi {
+    private final StudentService studentService;
+    @GetMapping
+    public ResponseEntity<List<StudentModel>> getStudents(){
+        List<StudentModel> studentList = studentService.getStudents();
+        return new ResponseEntity<>(studentList, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<StudentModel> getStudentById(@PathVariable long id){
+        Optional<StudentModel> student = studentService.getStudentById(id);
```

Comment:
Zamiast tu zwracać optionala i później sprawdzać czy obiekt `isPresent()` to można poziom niżej to obsłużyć i tam wyrzucić jakiś wyjątek - taka luźna sugestia

Reply:
`StudentModel student = studentService.getStudentId(id);`